### PR TITLE
feat(cast): wallet keystore decrypting

### PR DESCRIPTION
**Pull request by rplusq**
## Motivation

After integrating `cast wallet import` and using it for a while, some users might want to get the private key for an account back through decryption of the keystone, so we're just making it easy for them in the same cli they created the keystore.

## Solution

Added an extra command `DecryptKeystore` to the `WalletSubcommands` enum, alongside the implementation where we verify the file exists, prompt for the password (they can also send it in clear text from the cli call if they wish to), decrypt the keystore to instantiate a `LocalWallet` and then make it an alloy `B256` so it formats properly in the print.

### CODE REVIEW

**Review by mattsse — CHANGES_REQUESTED** [review]
supportive, could we can a casttest! for this as well?
ideally run both commands in sequence

**Comment by rplusq** [comment]
Ready for review @mattsse

File: crates/cast/bin/cmd/wallet/mod.rs [hunk 1]
```diff
@@ -151,6 +151,22 @@ pub enum WalletSubcommands {
...
     #[command(name = "derive-private-key", visible_aliases = &["--derive-private-key"])]
     DerivePrivateKey { mnemonic: String, mnemonic_index: Option<u8> },
+
+    /// Decrypt a keystore file to get the private key
+    #[command(name = "decrypt-keystore", visible_alias = "dk")]
+    DecryptKeystore {
+        /// The name for the account in the keystore.
+        #[arg(value_name = "ACCOUNT_NAME")]
+        account_name: String,
+        /// If provided, keystore will be saved here instead of the default keystores directory
```
Comment (Evalir) [inline crates/cast/bin/cmd/wallet/mod.rs:161]: I think, for this command, this comment is innacurate?

**Review by mattsse — APPROVED** [review]
lgtm, pending doc nit

File: crates/cast/bin/cmd/wallet/mod.rs (hunk 1 above)
Comment (rplusq) [reply crates/cast/bin/cmd/wallet/mod.rs:161]: You're definitely right, missed that. Fixed now.

**Comment by rplusq** [comment]
Fixed @Evalir

**Review by mattsse — APPROVED** [review]

**Pull request merged**